app/validators/tags_validator.rb

```diff
@@ -22,4 +27,13 @@ def tag_values_are_arrays(tags)
       end
     end
   end
+
+  def invalid_formatted_tags(tags)
+    invalid = tags.select do |_key, tag_values|
+      tag_values.flat_map(&:last)
+        .any? { |tag| !tag.to_s.match?(/\A[a-zA-Z0-9-_ \/"\'\/{\}\]\[]*\z/) }
```

Comment:
It looks like this regex permits quotes in strings. Is that something we want to permit? If so I think it'd be worth adding a test for this case.

If we don't want quotes, would this work? `(/\A[a-zA-Z0-9-_{}\]\[]*\z/)`

Reply:
There is a test in spec/queries/subscriber_lists_by_criteria_query_spec.rb line 65, which includes quotes. That's the reason I permitted quotes. I'm not sure if this is something we want to change?

Reply:
Ah good spot! In that case I think you've got it right 👍 

We allow some pretty funky stuff in tags, this'd be a valid tag: `[]{}/" '123-Abc`.

Reply:
I don't know if that test means we need to support those things. I wrote that test originally and I expect I was just unit testing places where I thought this was liable to break. 

If we don't have use of those type of tags in the wild we can kill off that test.

I just did a quick experiment and it looks like we don't have any outside alpha numeric characters, numbers, dashes, underscores and forward slash:

```
irb(main):042:0> SubscriberList.where("tags::text != '{}'::text").pluck(:tags).inject([]) { |memo, tags| memo + tags.values.last.flat_map(&:last) }.uniq.all? { |a| a.match?(/\A[a-z0-9\-_\/]*\Z/) }
=> true
```
(Ran on staging)

Reply:
If we don't currently use them, it probably means we don't have a need for them. I'd say it's safe to make the regex simpler and remove the chars that are not needed.

Reply:
Thanks for looking those up Kevin! That sounds good to me. @MahmudH do you want to try using `(/\A[a-zA-Z0-9\-_{}\]\[]*\z/)` and updating/removing those other tests?

Reply:
I'd suggest you can reduce this further to not have `{}` and `[]` since we don't have known usage of them or plans to use them.